client/client.go

```diff
@@ -112,6 +113,39 @@ func (client *Client) doSync(method, path string, query url.Values, body io.Read
 	return nil
 }
 
+type asyncResult struct {
+	Resource string `json:"resource"`
+}
+
+func (client *Client) doAsync(method, path string, query url.Values, body io.Reader) (string, error) {
+	var rsp response
+
+	if err := client.do(method, path, query, body, &rsp); err != nil {
+		return "", fmt.Errorf("failed to communicate with server: %v", err)
```

Comment:
s/failed to/cannot/

More generically I propose we: `failed to`/`unable to`/`error while`/`found error`/`can't`/ => `cannot`